--- Audio-Ecasound-0.2/Makefile.PL.orig	Wed Jun 27 15:59:50 2001
+++ Audio-Ecasound-0.2/Makefile.PL	Sat Feb 15 11:46:12 2003
@@ -5,7 +5,7 @@
     VERSION_FROM    => 'Ecasound.pm', 
     PREREQ_PM        => {}, # e.g., Module::Name => 1.1
     # the -lkvutils shouldn't be required
-    LIBS            => ['-lecasoundc -lkvutils'], 
+    LIBS            => ['-lecasoundc'], 
     INC                => '', 
     ABSTRACT_FROM    => 'Ecasound.pm', 
     AUTHOR            => 'Brad Bowman <eci-perl@bowman.bs>', 
@@ -38,13 +38,13 @@
             }
             
             # ecasound < 2.0.1 didn't have reentrant C interface
-            my $eca_version = `ecasoundc-config --version`;
+            my $eca_version = `libecasoundc-config --version`;
             if(!defined($eca_version)) {
                 warn "Running ecasoundc-config failed: $!\n";
                 
                 print <<EOT;
-The ecasoundc-config program is used to find the version of the
-ecasoundc library.  You need 2.0.1 or higher.
+The libecasoundc-config program is used to find the version of the
+libecasoundc library.  You need 2.0.1 or higher.
 EOT
                 print "Proceed anyway? ";
                 for(print "(yn)\n"; $_ = <STDIN>; print "(yn)?\n") {
@@ -55,7 +55,7 @@
                 chomp $eca_version;
                 # Compare versions pre v-strings 
                 my @ev = split /\./, $eca_version;
-                unless($ev[0] >= 2 && $ev[1] >= 0 && $ev[2] >= 1) {
+                unless($ev[0] >= 2 && ($ev[1] > 0 || $ev[1] == 0 && $ev[2] >= 1)) {
                     print <<EOT;
 Version 2.0.1 of ecasound is required, you have $eca_version.
 EOT
--- Audio-Ecasound-0.2/Ecasound.xs.orig	Wed Jun 27 15:59:49 2001
+++ Audio-Ecasound-0.2/Ecasound.xs	Sat Feb 15 11:45:10 2003
@@ -2,7 +2,7 @@
 #include "perl.h"
 #include "XSUB.h"
 
-#include <ecasound/ecasoundc.h>
+#include <libecasoundc/ecasoundc.h>
 
 MODULE = Audio::Ecasound        PACKAGE = Audio::Ecasound
 
--- Audio-Ecasound-0.2/Ecasound.pm.orig	Wed Jun 27 15:59:50 2001
+++ Audio-Ecasound-0.2/Ecasound.pm	Sat Feb 15 11:54:52 2003
@@ -193,6 +193,8 @@
         $self->last_long_integer();
     } elsif($type eq '') {
         return ''; # false but defined
+    } elsif($type eq '-') {
+	return '';
     } elsif($type eq 'S') {
         my $count = $self->last_string_list_count();
         # differentiate from () err when ambiguous
--- Audio-Ecasound-0.2/test.pl.orig	Wed Jun 27 15:59:49 2001
+++ Audio-Ecasound-0.2/test.pl	Sat Feb 15 12:22:54 2003
@@ -30,7 +30,7 @@
 ok eci_last_type() => 'f'; 
 ok eci_last_float() => 0;
 eci_command_float_arg('cs-set-length', 2.0);
-ok eci_last_type() => '';
+ok eci_last_type() => '-';
 eci_command('start'); 
 ok eci_error() => 1;
 ok eci_last_error() => qr/chainsetup cannot be connected/;
@@ -43,11 +43,11 @@
 ok ref($eh) => 'eci_handle_t';
 eci_command_r($eh,'status'); 
 ok eci_last_type_r($eh) => 's';
-ok eci_last_string_r($eh) => qr/Engine status/;
+ok eci_last_string_r($eh) => qr/Chainsetup status/;
 eci_cleanup_r($eh);
 
 # simple (test 18)
-ok eci('status') => qr/Engine status/;
+ok eci('status') => qr/Chainsetup status/;
 
 ok on_error() => 'warn'; # default val
 ok on_error('') => '';
@@ -72,17 +72,17 @@
 ok $e->on_error() => 'die'; # diverge from class
 $e->errmsg('');
 eval { $e->eci('start') };
-ok $@ => qr/no chainsetup connected/;
+ok $@ => qr/No chainsetup connected/;
 eval { $e->eci('cs-set-length'); };
 ok $@ => qr/argument omitted/;
 ok $e->errmsg() => qr/argument omitted/;
-ok $e->eci('status') => qr/Engine status/;
+ok $e->eci('status') => qr/Chainsetup status/;
 $e->eci('cs-add c1');
 $e->eci('cs-set-length', 5.0);
 ok $e->eci('cs-get-length') => 5;
 
 # comments added
-ok eci('status # a comment') => qr/Engine status/;
+ok eci('status # a comment') => qr/Chainsetup status/;
 # multiline commands
 ok eci("cs-set-length 6
         cs-get-length") => 6;
