--- Audio-Ecasound-0.2/Makefile.PL.orig	Wed Jun 27 15:59:50 2001
+++ Audio-Ecasound-0.2/Makefile.PL	Sun Aug  4 19:36:40 2002
@@ -5,7 +5,7 @@
     VERSION_FROM    => 'Ecasound.pm', 
     PREREQ_PM        => {}, # e.g., Module::Name => 1.1
     # the -lkvutils shouldn't be required
-    LIBS            => ['-lecasoundc -lkvutils'], 
+    LIBS            => ['-lecasoundc2 -lkvutils3'], 
     INC                => '', 
     ABSTRACT_FROM    => 'Ecasound.pm', 
     AUTHOR            => 'Brad Bowman <eci-perl@bowman.bs>', 
@@ -38,7 +38,7 @@
             }
             
             # ecasound < 2.0.1 didn't have reentrant C interface
-            my $eca_version = `ecasoundc-config --version`;
+            my $eca_version = `libecasoundc2-config --version`;
             if(!defined($eca_version)) {
                 warn "Running ecasoundc-config failed: $!\n";
                 
@@ -55,7 +55,7 @@
                 chomp $eca_version;
                 # Compare versions pre v-strings 
                 my @ev = split /\./, $eca_version;
-                unless($ev[0] >= 2 && $ev[1] >= 0 && $ev[2] >= 1) {
+                unless($ev[0] >= 2 && ($ev[1] > 0 || $ev[1] == 0 && $ev[2] >= 1)) {
                     print <<EOT;
 Version 2.0.1 of ecasound is required, you have $eca_version.
 EOT
--- Audio-Ecasound-0.2/Ecasound.xs.orig	Wed Jun 27 15:59:49 2001
+++ Audio-Ecasound-0.2/Ecasound.xs	Sun Aug  4 19:37:16 2002
@@ -2,7 +2,7 @@
 #include "perl.h"
 #include "XSUB.h"
 
-#include <ecasound/ecasoundc.h>
+#include <libecasoundc2/ecasoundc.h>
 
 MODULE = Audio::Ecasound        PACKAGE = Audio::Ecasound
 
